src/gui/widgets/LcdWidget.cpp

```diff
@@ -79,16 +82,15 @@ LcdWidget::~LcdWidget()
 void LcdWidget::setValue( int value )
 {
 	QString s = m_textForValue[value];
-	if( s.isEmpty() )
+	if (s.isEmpty())
 	{
 		s = QString::number( value );
-		// TODO: if pad == true
-		/*
-		while( (int) s.length() < m_numDigits )
-		{
-			s = "0" + s;
+		if (m_leadingZero) {
+			while (s.length() < m_numDigits)
+			{
+				s = "0" + s;
+			}
```

Comment:
It's short enough this should be fine I think.
```suggestion
			while (s.length() < m_numDigits) { s = "0" + s; }
```

Reply:
Or ditch the loop and use

``int pad = max(0, m_numDigits - s.length());``
``s.insert(0, pad, '0');``

Or if s is a QString (can't tell since diff won't load), maybe there's a similar function.

Reply:
Yeah, that may have been cleaner and would all fit on one line with the surrounding `if`. But I already commited Veratil's suggestion, so :man_shrugging: :))